packages/remark-extend/src/remarkExtend.js

```diff
@@ -0,0 +1,291 @@
+// unified works my modifying the original passed node
+/* eslint-disable no-param-reassign */
+const visit = require('unist-util-visit');
+const { select } = require('unist-util-select');
+const unified = require('unified');
+const markdown = require('remark-parse');
+const is = require('unist-util-is');
+
+function addTask(file, newAction) {
+  if (!file.data.remarkExtend) {
+    file.data.remarkExtend = [];
+  }
+  file.data.remarkExtend.push(newAction);
+}
+
+function findReplacementTasks(tree, file) {
+  visit(tree, 'code', node => {
+    if (node.lang === 'js' && node.meta && node.meta.startsWith('::replaceFrom')) {
+      const startSelector = node.meta.substring(
+        node.meta.indexOf("('") + 2,
+        node.meta.indexOf("')"),
+      );
+      addTask(file, {
+        action: 'replaceFrom',
+        startSelector,
+        jsCode: node.value,
+      });
+    }
+    if (node.lang === 'js' && node.meta && node.meta.startsWith('::replaceBetween')) {
+      const startSelector = node.meta.substring(
+        node.meta.indexOf("('") + 2,
+        node.meta.indexOf("',"),
+      );
+      const endSelector = node.meta
+        .substring(node.meta.indexOf("',") + 4, node.meta.indexOf("')"))
```

Comment:
this is on purpose "fragile"... you need to write it as it's intended or it will throw that it can't find the selector and then it will give you what it sees as the selector

e.g. you also can't use `"`  like `::replaceBetween("`

I think that is good enough for now... if we ever need to have a full "processing" her... e.g. actually making kind of an "AST" of this extra block then we can revisit

Reply:
Aah okay if we throw a useful error I'm fine with the fragility. I agree to make it properly nice we'd have to use something better than "'," and regex, e.g. AST, but that could be for way later.